### PR TITLE
newfunctions

### PR DESCRIPTION
**Pull request by samardhk**
added calorie function that spits out macros as well.

### CODE REVIEW

File: calculation.js [hunk 1]
```diff
@@ -3,7 +3,12 @@ function calculateBMI() {
     var feet = document.getElementById('feet').value;
     var inches = document.getElementById('inches').value || 0; // default to 0 if inches are not provided
     var pounds = document.getElementById('pounds').value;
-
+    var age = document.getElementById('age').value;
```
Comment (atul-jalan) [inline calculation.js]: These should all be either 'const' or 'let'.

File: calculation.js [hunk 2]
```diff
@@ -14,3 +19,80 @@ function calculateBMI() {
 
     document.getElementById('result').innerHTML = `Your BMI is ${bmi}`;
 }
+
+
+       
+function TDEE() {
```
Comment (atul-jalan) [inline calculation.js:36]: what does TDEE mean?

File: calculation.js [hunk 3]
```diff
@@ -14,3 +19,80 @@ function calculateBMI() {
...
+    var fitnessvalue = document.getElementById("fitness").value;
+    if (!feet || !pounds || !age || !updatedvalue ||!fitnessvalue) {
+        alert("Please complete all fields.");
+        return; 
+    }
+    
+    
+
+    var inches = document.getElementById('inches').value || 0; 
+    var totalInches = (+feet * 12) + +inches;
```
Comment (atul-jalan) [inline calculation.js]: what does `+feet` mean?

File: calculation.js [hunk 4]
```diff
@@ -14,3 +19,80 @@ function calculateBMI() {
...
+        return; 
+    }
+    
+    
+
+    var inches = document.getElementById('inches').value || 0; 
+    var totalInches = (+feet * 12) + +inches; 
+    var heightInCM = totalInches * 2.54; 
+    var weightInKilos = pounds * 0.453592;
+    var bmr = (10 * weightInKilos) + (6.25 * heightInCM) - (5 * age) + 5;
```
Comment (atul-jalan) [inline calculation.js]: Generally in programming, we declare variables for constant values.

So at the top of the `calculation.js` file, you'd declare the following:

```
const CENTIMETERS_PER_INCH = 2.54;
const KILOS_PER_POUND = .453592;
...
```

And then the actual calculation would be:

```
const heightInCM = totalInches * CENTIMETERS_PER_INCH
...
```

File: calculation.js [hunk 5]
```diff
@@ -14,3 +19,80 @@ function calculateBMI() {
...
+
+    var sed = 1.2;
+    var light = 1.375;
+    var mod = 1.55;
+    var very = 1.725;
+    var sup = 1.9;
```
Comment (atul-jalan) [inline calculation.js]: I would delcare this as a constant at the top of the `calculation.js` file. in the following way:

```
const ACTIVITY_LEVEL_OPTIONS = {
    sed: 1.2,
    light: 1.375,
    ...
}
```

You can access the values in the following way:

```
const tdee = (bmr * ACTIVITY_LEVEL_OPTIONS.sed) - 500
```

File: calculation.js [hunk 6]
```diff
@@ -14,3 +19,80 @@ function calculateBMI() {
...
+
+    
+
+    
+    var tdee;
+    if (updatedvalue=="Sedentary" && fitnessvalue =="lose") {
+        var tdee = (bmr * sed) - 500;
```
Comment (atul-jalan) [inline calculation.js]: So here's an example of why it's useful to declare constant variables.

a) You use the number 500 a dozen times throughout this code. Changing it means having to change it everywhere it's used.
b) It's not clear what the 500 value represents to someone reading this code. It takes a second to understand that it represents the daily calories gain/loss.

Think of a descriptive term for the `500` value, then declare at the top of the file, and use that variable everywhere.

**Review by atul-jalan — CHANGES_REQUESTED** [review]

File: calculation.js (hunk 2 above)
Comment (samardhk) [reply calculation.js:36]: total daily energy expenditure (so like total calories to consume everyday)

File: calculation.js (hunk 3 above)
Comment (samardhk) [reply calculation.js]: i think it just converts it into a value so things like 5.5 can be added correctly and if letters are input itll say invalid

File: calculation.js [hunk 7]
```diff
@@ -1,9 +1,20 @@
+const centimeterperinch = 2.54;
+const kilosperpound = 0.453592;
+const dailycalgainloss = 500
+const activityleveloptions = {
```
Comment (atul-jalan) [inline calculation.js]: When you declare constant value variables (meaning variables that represent some known static value), you want to distinguish them from other variables. For example, you should declare `centimeterperinch` in the following way:

```
const CENTIMETERS_PER_INCH = 2.54
```

Then, as you reference this variable in your code, it's clear that this is a constant value variable.

**Review by atul-jalan — APPROVED** [review]
lgtm. One change requested.

Also please audit your code and remove `var` everywhere and replace with `let` or `const`.
